src/framework/network/internal/networkconfiguration.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "networkconfiguration.h"
+
+#include "global/configreader.h"
+
+using namespace muse;
+using namespace muse::network;
+
+void NetworkConfiguration::init()
+{
+    m_userAgentConfig = ConfigReader::read(":/configs/user_agent.cfg");
+}
+
+RequestHeaders NetworkConfiguration::headers(const std::string& userAgentTemplate) const
+{
+    RequestHeaders headers;
+
+    headers.knownHeaders[QNetworkRequest::UserAgentHeader] = makeUserAgent(userAgentTemplate);
+
+    return headers;
+}
+
+QString NetworkConfiguration::makeUserAgent(const std::string& tpl) const
```

Comment:
as I understand it, it is not the template itself that is transferred here, but only the name
tpl -> `name` or `tplName`